### DISCUSSION TITLE
2Ch wall switch module with binding as source

### DISCUSSION THREAD

**Discussion by Flamingi** [Koenkk/zigbee2mqtt#18028] [Devices]
Hi all,
I'm searching for a wall switch module usable in Europe (230V/50Hz, those mini modules 40x40x20mm). It should be wired up with a neutral wire, i.e. no battery, and support two inputs. I want to bind the module to my smart bulbs, so it must have two enpoints with genOnOff Output clusters. I feel like those are no crazy requirements, but I'm totally stuck in my search and so far I've bought multiple different TuYa modules which can't be used for binding. I found a few modules that should work (Ubisys S2, Gewiss GWA 1522), but those are 100€ each, which I'm not willing to spend. TuYa modules can be had for as little as 4€ on Aliexpress, but I'd say max budget would be around 20€ per module.

I'd appreciate any suggestions which module might work :)

**Comment by andrazek** [comment]
Have you found something? Sunreicher seems to have such switch modules that support direct binding.

**Reply by Flamingi** [reply]
I did find some other manufacturers which seemingly support binding, but they all were 100€ again, which I‘m not willing to spend. However, there is a project writing custom firmware for TuYa Zigbee devices: https://github.com/doctor64/tuyaZigbee. At the time there was nothing for the switches, but maybe that changed until now, there seems to be activity in that project still.
